src/validators/RecordValidator.ts

```diff
@@ -0,0 +1,36 @@
+import type { IConstraint } from '../constraints/base/IConstraint';
+import { ValidationError } from '../lib/errors/ValidationError';
+import { Result } from '../lib/Result';
+import { BaseValidator } from './imports';
+
+export class RecordValidator<T> extends BaseValidator<Record<string, T>> {
+	private readonly validator: BaseValidator<T>;
+
+	public constructor(validator: BaseValidator<T>, constraints: readonly IConstraint<Record<string, T>>[] = []) {
+		super(constraints);
+		this.validator = validator;
+	}
+
+	protected override clone(): this {
+		return Reflect.construct(this.constructor, [this.validator, this.constraints]);
+	}
+
+	protected handle(value: unknown): Result<Record<string, T>, ValidationError | AggregateError> {
+		if (!(typeof value === 'object')) {
+			return Result.err(new ValidationError('RecordValidator', 'Expected a record', value));
```

Comment:
Shouldn't this say it expects an object? JS-only users will be more familiar with "object" than with this TypeScript construct.

Reply:
I figured if they didn't know what a record was, they wouldn't be using the record validator, so most people who came across this error would know what a record was and it would help distinguish it from the object validator a bit more.

Reply:
Problem is, libraries may use this library, and they may use record, something developers using said library wouldn't see.